tests/test_utils/test_schemaview.py

```diff
@@ -736,6 +738,150 @@ def test_in_schema(schema_view_with_imports: SchemaView) -> None:
+            "schema": "http://schema.org/",
+            "rdfs": "https://www.w3.org/2000/01/rdf-schema#",
+            "prov": "http://www.w3.org/ns/prov#",
+            "GSSO": "http://purl.obolibrary.org/obo/GSSO_",
+            "famrel": "https://example.org/FamilialRelations#",
+            "bizcodes": "https://example.org/bizcodes/",
+            "skos": "http://www.w3.org/2004/02/skos/core#",
+            "P": "http://example.org/P/",
+            "ROR": "http://example.org/ror/",
+            "CODE": "http://example.org/code/",
+            "GEO": "http://example.org/geoloc/",
+        },
+    },
+}
+curi_maps_to_add = {
+    None: {
+        "exp": {},
+    },
+    "semweb_context": {
+        "text": "\ndefault_curi_maps:\n  - semweb_context\n",
+        "exp": {
+            "dc": "http://purl.org/dc/terms/",
+            "dcat": "http://www.w3.org/ns/dcat#",
+            "dcterms": "http://purl.org/dc/terms/",
+            "faldo": "http://biohackathon.org/resource/faldo#",
+            "foaf": "http://xmlns.com/foaf/0.1/",
+            "idot": "http://identifiers.org/",
+            "oa": "http://www.w3.org/ns/oa#",
+            "owl": "http://www.w3.org/2002/07/owl#",
+            "prov": "http://www.w3.org/ns/prov#",
+            "rdf": "http://www.w3.org/1999/02/22-rdf-syntax-ns#",
+            "rdfs": "http://www.w3.org/2000/01/rdf-schema#",
+            "void": "http://rdfs.org/ns/void#",
+            "xsd": "http://www.w3.org/2001/XMLSchema#",
+            "oboInOwl": "http://www.geneontology.org/formats/oboInOwl#",
+        },
+    },
+}
+# Prefixes from imports only show up if a function that generates the imports closure is run.
+imports_to_add = {
+    None: {"exp": {}, "imported": {}},
+    "linkml_types": {
+        "text": "\nimports:\n  - linkml:types\n",
+        "exp": {},
+        "imported": {
+            "linkml": "https://w3id.org/linkml/",
+            "xsd": "http://www.w3.org/2001/XMLSchema#",
+            "shex": "http://www.w3.org/ns/shex#",
+            "schema": "http://schema.org/",
+        },
+    },
+}
+
+
+@pytest.mark.parametrize("prefix", prefixes_to_add.keys())
+@pytest.mark.parametrize("curi_map", curi_maps_to_add.keys())
+@pytest.mark.parametrize("imports", imports_to_add.keys())
+@pytest.mark.parametrize("run_imports", [True, False])
+def test_namespaces(prefix: str | None, curi_map: str | None, imports: str | None, run_imports: bool) -> None:
+    """Check that the `namespaces` function correctly loads the prefix <==> URL mapping.
+
+    Note: the `namespaces` function does not check whether a prefix already exists in the mapping and whether
+    the existing value is the same as the incoming value; the incoming value always overwrites any existing value.
+    To test this, the RDFS URL is "http://" in the "semweb_context" mapping and "https://" in the "prefixes" section.
+    Whenever the prefixes section is present, it will override the value from semweb_context.
```

Comment:
Is it really so on purpose? Is it somewhere clearly documented?

Reply:
> I don't know if the current MO is deliberate or not; I would guess that it was implemented without considering the fact that you could easily have conflicting values for the same prefix and the order in which they are specified in the schema file(s) affects what mappings of prefix to URL you end up with.

I would consider such a behavior without any documentation a flaw... and would probably report a bug for it. Would you agree?

Reply:
Yes -- I would have the schemaview raise an error if there are conflicts (or have a strict mode that throws errors).

Unfortunately, on my explorations through SchemaView land, I have found numerous cases where errors like this are silently ignored or where SchemaView benevolently assumes that schemas don't contain errors, contradictions, etc.  Just expanding some tests in response to your comments, I've discovered some more assumptions made by SchemaView that have an effect on functionality if a schema contains errors or missing data. I know you have also found issues that affect functionality and that aren't covered by linter or other tests. Noting or fixing the issues starts feeling like an impossible task...

Reply:
I've also identified some [inconsistencies WRT inlining](https://github.com/linkml/linkml/issues/2822). Additionally SchemaView and SchemaLoader handle them differently due to the [lack of sane defaults in SchemaLoader](https://github.com/linkml/linkml/issues/2813) 🤦